solana/pyth2wormhole/client/src/main.rs

```diff
@@ -301,56 +272,181 @@ async fn handle_attest(
         Duration::from_millis(attestation_cfg.min_rpc_interval_ms),
     ));
 
-    let message_q_mtx = Arc::new(Mutex::new(P2WMessageQueue::new(
-        Duration::from_millis(attestation_cfg.min_msg_reuse_interval_ms),
-        attestation_cfg.max_msg_accounts as usize,
-    )));
+    // The loop below attempts to crank attestations without
+    // interruption. This is achieved by spinning up an up-to-date
+    // symbol set before letting go of the previous one. This Option
+    // preserves the state of potential ongoing scheduling jobs. The
+    // hash is sensitive to both attestation_cfg and on-chain config
+    // changes.
+    let mut old_sched_futs_state: Option<(JoinHandle<_>, u64)> = None; // (old_futs_handle, old_config_hash)
 
-    // Create attestation scheduling routines; see attestation_sched_job() for details
-    let mut attestation_sched_futs = batches.into_iter().map(|(batch_no, batch)| {
-        attestation_sched_job(
-            batch,
-            batch_no,
-            batch_count,
-            n_retries,
-            retry_interval,
-            daemon,
-            rpc_cfg.clone(),
-            p2w_addr,
-            config.clone(),
-            Keypair::from_bytes(&payer.to_bytes()).unwrap(),
-            message_q_mtx.clone(),
-        )
-    });
+    // This loop governs mapping account reloads in daemon mode. It is
```

Comment:
this function is long and the logic is hard to understand because it supports all these different options around daemon / no daemon. 

I think it would be good to refactor this by extracting out helper functions and then possibly having two separate top-level functions for daemon / non-daemon mode. Each of those functions could use the helpers to avoid code duplication.